### PR TITLE
Revisit environment doc

### PR DESCRIPTION
**Pull request by ianmalott**
## Release Playbook
- [x] Rebase against master
- [x] Code review
- [ ] Merge into master
- [ ] Verify changes at http://site-staging.convox.com
- [ ] Promote release on `site-production` in the `convox/production` Rack

### CODE REVIEW

File: _management/environment.md [hunk 1]
```diff
@@ -3,41 +3,86 @@ title: "Environment"
...
+    environment:
+      - SECRET_KEY    # ignored by deployed apps; only used locally
+      - FOO=default   # override with `convox env set FOO=newvalue`
+```
+
+## Sensitive Information
+
+Since `docker-compose.yml` will typically be under version control, we recommend not setting sensitive default values there.
+
+We also recommend leaving your `.env` file out of version control to protect against leaking any sensitive information.
```
Comment (mattmanning) [inline _management/environment.md]: Please also recommend here that `.env` be included in `.dockerignore`

👍 to publish after that.

Comment (ianmalott) [inline _management/environment.md]: Yup, nice catch.

File: _management/environment.md [hunk 2]
```diff
@@ -3,41 +3,86 @@ title: "Environment"
...
-    - OTHER_VAR=foo
+SECRET_KEY=xyzzy
+FOO=bar
 ```
 
-### Local
+#### docker-compose.yml
 
-When running your application in `convox start` you can set values for your application's environment in a `.env` file:
+`convox start` will always read `.env`, but the environment variables set there will not be passed along to your application's containers unless you declare them in `docker-compose.yml`.
```
Comment (mwarkentin) [inline _management/environment.md]: Maybe provide the rationale for this decision here?

Comment (ianmalott) [inline _management/environment.md]: Thanks for the feedback. Looks like I just missed you. We're discussing some changes that could make local and deployed environment management more similar, so this doc could be getting overhauled soon. If we punt, I'll circle back.

**Pull request merged**